android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/adapters/NestedAdapter.java

```diff
@@ -0,0 +1,179 @@
+package com.tavultesoft.kmea.data.adapters;
+
+import android.content.Context;
+import android.database.DataSetObserver;
+import android.widget.ArrayAdapter;
+
+import androidx.annotation.NonNull;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * An ArrayAdapter subclass that uses another ArrayAdapter (or subclass) as its data source,
+ * propagating relevant changes across both to maintain a 'linked'/'nested' state.
+ * Also permits use of an AdapterFilter to allow using a filtered view of that source.
```

Comment:
When paired with the `Dataset` class and its nested classes, this new `NestedAdapter` is designed to allow re-use of a single common datastructure across all of the settings UI menus - even if that base data structure must be accessed by semi-global / `static` singleton on the `CloudRepository` class (for cloud stuff) and from the `KeyboardPickerActivity` (for installed resources).

This class then serves as the heart of the event-messaging functionality needed to drive our desired UX in the Settings menus.